File: jump-ant-tasks/com/ollix/jump/ant/JythonCompiler.java
/****************************************************************************
**
** Copyright (c) 2009 Ollix. All rights reserved.
**
** This file is part of Jump.
**
** Jump is free software: you can redistribute it and/or modify it under the
** terms of the GNU General Public License as published by the Free Software
** Foundation, either version 3 of the License, or any later version.
**
** Jump is distributed in the hope that it will be useful, but WITHOUT ANY
** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
** details.
**
** You should have received a copy of the GNU General Public License along
** with Jump. If not, see <http://www.gnu.org/licenses/>.
**
****************************************************************************/

package com.ollix.jump.ant;

import org.apache.tools.ant.BuildException;
import org.apache.tools.ant.Task;
import com.ollix.jump.JythonFactory;
import com.ollix.jump.ant.JythonCompilerType;

public class JythonCompiler extends Task
{
    private JythonCompilerType jythonCompiler;

    public JythonCompiler() {
        JythonFactory supportSitePackages = new JythonFactory(
            "jump_jython_factory.syspath", "support_site_packages");
        supportSitePackages.call(true);

        JythonFactory factory = new JythonFactory(
            supportSitePackages.getState(), JythonCompilerType.class,
            "jump.ant.jython_compiler", "JythonCompiler");

        jythonCompiler = (JythonCompilerType) factory.init();
    }

    public void execute() throws BuildException {
        jythonCompiler.execute();
    }

    public void setDestDir(String destDir) {
        jythonCompiler.setDestDir(destDir);
    }

    public void setFullPackages(String packages) {
        jythonCompiler.setFullPackages(packages);
    }
}

File: jump/ant/jython_compiler.py
#!/usr/bin/env python
# encoding: utf-8
"""
jython_compiler.py

Created by Olli Wang (olliwang@ollix.com) on 2009-11-18.
Copyright (c) 2009 Ollix. All rights reserved.

Jump is free software: you can redistribute it and/or modify it under the
terms of the GNU General Public License as published by the Free Software
Foundation, either version 3 of the License, or any later version.

Jump is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
A PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with Jump. If not, see <http://www.gnu.org/licenses/>.
"""

import os
import sys
import shutil

from pylibtracer.libtracer import LibTracer

from com.ollix.jump.ant import JythonCompilerType


class JythonCompiler(JythonCompilerType):
    """Copies and compiles required Python modules to specified directory."""

    def __init__(self, dest_dir=None, packages=None):
        self.dest_dir = dest_dir
        self.packages = packages

    def execute(self):
        if self.packages:
            packages = [p.strip() for p in self.packages.split(',')]
        else:
            packages = []

        # Copy dependent files to specified directory
        sys.path.append(self.dest_dir)
        lib_tracer = LibTracer(full_packages=packages, quiet=True)
        count = lib_tracer.copy_to(self.dest_dir)
        if count:
            print 'Copy %d files to %s' % (count, self.dest_dir)

    def setDestDir(self, dest_dir):
        self.dest_dir = dest_dir

    def setFullPackages(self, packages):
        self.packages = packages
